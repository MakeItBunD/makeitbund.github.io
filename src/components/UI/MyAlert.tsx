import React from 'react'

function MyAlert() {
    return (
        <div className="alert">
            <h2 className="alert__title">Товар добавлен в корзину</h2>
        </div>
    )
}

export default MyAlert