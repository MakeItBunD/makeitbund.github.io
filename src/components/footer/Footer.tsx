import React from 'react'
import logoIcon from 'D:/Курсы/shop/src/icons/logo.svg'
import MyInput from '../UI/MyInput'
import arrowRightIcon from 'D:/Курсы/shop/src/icons/arrow_right.svg'
import downloadIcon from 'D:/Курсы/shop/src/icons/download.svg'
import whatsupIcon from 'D:/Курсы/shop/src/icons/whatsup.svg'
import telegramIcon from 'D:/Курсы/shop/src/icons/telegram.svg'
import visaIcon from 'D:/Курсы/shop/src/icons/visa.svg'
import mastercardIcon from 'D:/Курсы/shop/src/icons/mastercard.svg'
import MyButton from '../UI/MyButton'

function Footer() {
    return (
        <div className="footer">
            <div className="container">
                <div className="footer__container">
                    <div className="footer__column">
                        <img src={logoIcon} alt="" className="footer__logo" />
                        <p className="footer__text">Компания «Султан» — снабжаем<br />розничные магазины товарами<br />"под ключ" в Кокчетаве и Акмолинской<br />области</p>
                        <p className="footer__text footer__text_subscribe">Подпишись на скидки и акции</p>
                        <div className="footer__input-container">
                            <MyInput 
                                className='footer__input'
                                placeholder='Введите ваш E-mail'
                                type='text'
                            />
                            <MyButton className="footer__button">
                                <img src={arrowRightIcon} alt="" />
                            </MyButton>
                        </div>
                    </div>

                    <div className="footer__column">
                        <h2 className="footer__title">Меню сайта:</h2>
                        <p className="footer__link">О компании</p>
                        <p className="footer__link">Доставка и оплата</p>
                        <p className="footer__link">Возврат</p>
                        <p className="footer__link">Контакты</p>
                    </div>

                    <div className="footer__column">
                        <h2 className="footer__title">Категории:</h2>
                        <p className="footer__link">Бытовая химия</p>
                        <p className="footer__link">Косметика и гигиена</p>
                        <p className="footer__link">Товары для дома</p>
                        <p className="footer__link">Товары для детей и мам</p>
                        <p className="footer__link">Посуда</p>
                    </div>

                    <div className="footer__column">
                        <h2 className="footer__title">Скачать прайс-лист:</h2>
                        <MyButton className='footer__price-button'>
                            <p>Прайс-лист</p>
                            <img src={downloadIcon} alt="" />
                        </MyButton>
                        <p className="footer__text footer__text_contact">Связь в мессенджерах:</p>
                        <div className="footer__social-media">
                            <div className="footer__media-link footer__media-link_whatsup">
                                <img src={whatsupIcon} alt="" />
                            </div>
                            <div className="footer__media-link footer__media-link_telegram">
                                <img src={telegramIcon} alt="" />
                            </div>
                        </div>
                    </div>

                    <div className="footer__column">
                        <h2 className="footer__title">Контакты:</h2>
                        <p className="footer__number">+7 (777) 490-00-91</p>
                        <p className="footer__working-time">время работы: 9:00-20:00</p>
                        <p className="footer__phone-order">Заказать звонок</p>
                        <p className="footer__mail">opt.sultan@mail.ru<br /><span>На связи в любое время</span></p>
                        <div className="footer__credit-cards">
                            <img src={visaIcon} alt="" className="footer__card"/>
                            <img src={mastercardIcon} alt="" className="footer__card"/>
                        </div>
                    </div>

                </div>
            </div>
        </div>
    )
}

export default Footer