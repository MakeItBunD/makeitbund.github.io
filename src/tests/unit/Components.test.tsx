import React from 'react';
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react';
import MyAlert from '../../components/UI/MyAlert';
import Footer from '../../components/footer/Footer';
import FooterMobile from '../../components/footer/FooterMoble';

describe('COMPONENTS TEST WITH SNAPSHOTS', () => {
    test('my-alert should not be changed', () => {
        // arrange
        render(<MyAlert />)
        const myAlert = screen.getByTestId('my-alert')

        // assert
        expect(myAlert).toMatchSnapshot()
    })

    test('footer should not be changed', () => {
        // arrange
        render(<Footer />)
        const footer = screen.getByTestId('footer-component')

        // assert
        expect(footer).toMatchSnapshot()
    })

    test('footerMobile should not be changed', () => {
        // arrange
        render(<FooterMobile />)
        const footerMobile = screen.getByTestId('mobile-footer-component')

        // assert
        expect(footerMobile).toMatchSnapshot()
    })
})
