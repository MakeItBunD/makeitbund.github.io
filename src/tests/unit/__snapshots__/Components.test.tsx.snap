// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS TEST WITH SNAPSHOTS footer should not be changed 1`] = `
<div
  class="footer"
  data-testid="footer-component"
>
  <div
    class="container"
  >
    <div
      class="footer__container"
    >
      <div
        class="footer__column"
      >
        <img
          alt=""
          class="footer__logo"
          src="logo_white.svg"
        />
        <p
          class="footer__text"
        >
          Компания «Султан» — снабжаем
          <br />
          розничные магазины товарами
          <br />
          "под ключ" в Кокчетаве и Акмолинской
          <br />
          области
        </p>
        <p
          class="footer__text footer__text_subscribe"
        >
          Подпишись на скидки и акции
        </p>
        <div
          class="footer__input-container"
        >
          <input
            class="footer__input input"
            data-testid="my-input"
            placeholder="Введите ваш E-mail"
            type="text"
            value=""
          />
          <button
            class="footer__button button"
          >
            <img
              alt=""
              src="arrow_right.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="footer__column"
      >
        <h2
          class="footer__title"
        >
          Меню сайта:
        </h2>
        <p
          class="footer__link"
        >
          О компании
        </p>
        <p
          class="footer__link"
        >
          Доставка и оплата
        </p>
        <p
          class="footer__link"
        >
          Возврат
        </p>
        <p
          class="footer__link"
        >
          Контакты
        </p>
      </div>
      <div
        class="footer__column"
      >
        <h2
          class="footer__title"
        >
          Категории:
        </h2>
        <p
          class="footer__link"
        >
          Бытовая химия
        </p>
        <p
          class="footer__link"
        >
          Косметика и гигиена
        </p>
        <p
          class="footer__link"
        >
          Товары для дома
        </p>
        <p
          class="footer__link"
        >
          Товары для детей и мам
        </p>
        <p
          class="footer__link"
        >
          Посуда
        </p>
      </div>
      <div
        class="footer__column"
      >
        <h2
          class="footer__title"
        >
          Скачать прайс-лист:
        </h2>
        <button
          class="footer__price-button button"
        >
          <p>
            Прайс-лист
          </p>
          <img
            alt=""
            src="download.svg"
          />
        </button>
        <p
          class="footer__text footer__text_contact"
        >
          Связь в мессенджерах:
        </p>
        <div
          class="footer__social-media"
        >
          <div
            class="footer__media-link footer__media-link_whatsup"
          >
            <img
              alt=""
              src="whatsup.svg"
            />
          </div>
          <div
            class="footer__media-link footer__media-link_telegram"
          >
            <img
              alt=""
              src="telegram.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="footer__column"
      >
        <h2
          class="footer__title"
        >
          Контакты:
        </h2>
        <p
          class="footer__number"
        >
          +7 (777) 490-00-91
        </p>
        <p
          class="footer__working-time"
        >
          время работы: 9:00-20:00
        </p>
        <p
          class="footer__phone-order"
        >
          Заказать звонок
        </p>
        <p
          class="footer__mail"
        >
          opt.sultan@mail.ru
          <br />
          <span>
            На связи в любое время
          </span>
        </p>
        <div
          class="footer__credit-cards"
        >
          <img
            alt=""
            class="footer__card"
            src="visa.svg"
          />
          <img
            alt=""
            class="footer__card"
            src="mastercard.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`COMPONENTS TEST WITH SNAPSHOTS footerMobile should not be changed 1`] = `
<div
  class="footer-mobile"
  data-testid="mobile-footer-component"
>
  <div
    class="container"
  >
    <div
      class="footer-mobile__container"
    >
      <div
        class="footer-mobile__row footer-mobile__row_header"
      >
        <img
          alt=""
          class="footer-mobile__logo"
          src="logo_white.svg"
        />
        <button
          class="footer-mobile__price-button button"
        >
          <p>
            Прайс-лист
          </p>
          <img
            alt=""
            src="download.svg"
          />
        </button>
      </div>
      <div
        class="footer-mobile__row footer-mobile__row_subscribe"
      >
        <p
          class="footer-mobile__text"
        >
          Компания «Султан» — снабжаем розничные магазины товарами "под ключ" в Кокчетаве и Акмолинской области
        </p>
        <p
          class="footer-mobile__text footer-mobile__text_subscribe"
        >
          Подпишись на скидки и акции
        </p>
        <div
          class="footer-mobile__input-container"
        >
          <input
            class="footer-mobile__input input"
            data-testid="my-input"
            placeholder="Введите ваш E-mail"
            type="text"
            value=""
          />
          <button
            class="footer-mobile__button button"
          >
            <img
              alt=""
              src="arrow_right.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="footer-mobile__row footer-mobile__row_links"
      >
        <div
          class="footer-mobile__column"
        >
          <h2
            class="footer-mobile__title"
          >
            Меню сайта:
          </h2>
          <p
            class="footer-mobile__link"
          >
            О компании
          </p>
          <p
            class="footer-mobile__link"
          >
            Доставка и оплата
          </p>
          <p
            class="footer-mobile__link"
          >
            Возврат
          </p>
          <p
            class="footer-mobile__link"
          >
            Контакты
          </p>
        </div>
        <div
          class="footer-mobile__column"
        >
          <h2
            class="footer-mobile__title"
          >
            Категории:
          </h2>
          <p
            class="footer-mobile__link"
          >
            Бытовая химия
          </p>
          <p
            class="footer-mobile__link"
          >
            Косметика и гигиена
          </p>
          <p
            class="footer-mobile__link"
          >
            Товары для дома
          </p>
          <p
            class="footer-mobile__link"
          >
            Товары для детей и мам
          </p>
          <p
            class="footer-mobile__link"
          >
            Посуда
          </p>
        </div>
      </div>
      <div
        class="footer-mobile__row footer-mobile__row_contacts"
      >
        <h2
          class="footer-mobile__title footer-mobile__title_contact"
        >
          Контакты:
        </h2>
        <div
          class="footer-mobile__item"
        >
          <div
            class="footer-mobile__column footer-mobile__column_contact"
          >
            <p
              class="footer-mobile__number"
            >
              +7 (777) 490-00-91
            </p>
            <p
              class="footer-mobile__working-time"
            >
              время работы: 9:00-20:00
            </p>
            <p
              class="footer-mobile__phone-order"
            >
              Заказать звонок
            </p>
          </div>
          <div
            class="footer-mobile__column"
          >
            <p
              class="footer-mobile__contact"
            >
              Связь в мессенджерах:
            </p>
            <div
              class="footer-mobile__social-media"
            >
              <img
                alt=""
                class="footer-mobile__media-link footer-mobile__media-link_whatsup"
                src="whatsup.svg"
              />
              <img
                alt=""
                class="footer-mobile__media-link footer-mobile__media-link_telegram"
                src="telegram.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer-mobile__row footer-mobile__row_cards"
      >
        <p
          class="footer-mobile__mail"
        >
          opt.sultan@mail.ru
          <br />
          <span>
            На связи в любое время
          </span>
        </p>
        <div
          class="footer-mobile__credit-cards"
        >
          <img
            alt=""
            class="footer-mobile__card"
            src="visa.svg"
          />
          <img
            alt=""
            class="footer-mobile__card"
            src="mastercard.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`COMPONENTS TEST WITH SNAPSHOTS my-alert should not be changed 1`] = `
<div
  class="alert"
  data-testid="my-alert"
>
  <h2
    class="alert__title"
  >
    Товар добавлен в корзину
  </h2>
</div>
`;
